planner:
  sot_type: "SoTStatic"
  tasks:
  - frame_id: base
    name: Base Position
    planner_type: BaseSingleWaypoint
    initial_pos: [0,0]
    target_pos:
    - 6.0
    - 0.0
    cruise_speed: 0.5
    tracking_err_tol: 0.4

controller:
  home: "default"
  sdf_collision_avoidance_enabled: False
  self_collision_avoidance_enabled: False
  static_obstacles_collision_avoidance_enabled: True
  collision_cbf_gamma: 0.5

  collision_constraint_type: "SignedDistanceConstraintCBF" # SignedDistanceConstraint or SignedDistanceConstraintCBF
  type: "MPC"
  collision_constraints_softend:
    self: False
    sdf: False
    static_obstacles: False
  acados:
    ocp_solver_options:
      nlp_solver_max_iter: 200
      qp_solver_warm_start: 1
    slack_enabled:
      x: True
      x_e: True
      h_0: True
      h: False
      h_e: False

simulation:
  static_obstacles:
    enabled: True

include:
  - package: "mm_run"
    path: "config/robot/thing.yaml"
  - package: "mm_run"
    path: "config/controller/MPC.yaml"
  - package: "mm_run"
    path: "config/sim/simulation.yaml"
  - package: "mm_run"
    path: "config/scene/simple_obstacles.yaml"
  - package: "mm_run"
    path: "config/map/SDF2D.yaml"

logging:
  log_dir: "2d_collision"
  log_level: 10 # 0 - not set, 10 - debug, 20 - info
